### DISCUSSION TITLE
Yolov5 inference image wrong colors

### DISCUSSION THREAD

**Discussion by unrue** [ultralytics/yolov5#12711] [General]
I'm running inference on Yolov5 trained net. Sometimes, some of the output image with bounding box has a strange blue dominant. Attach the original and the output. Have you ever experienced such behaviour? Thanks.

My code:

```
`for i in range(num_images):
    print("index: ", i, "image: ", inference_dir+'/'+img_list[i])
    im = cv2.imread(inference_dir+'/'+img_list[i])
    dimensions = im.shape
    results = model(im)
    results.save("img_"+str(i)+".jpg", save_dir='out_with_bb') 
    res_list.append(results.pandas().xyxy[0])
    img_dimensions_list.append(dimensions)
`
```

![1890 1372 --557778](https://github.com/ultralytics/yolov5/assets/51996285/32bbc9a5-ca40-4331-abc9-66918effdbeb)
![image0](https://github.com/ultralytics/yolov5/assets/51996285/20174b06-b6d9-4aa3-ba77-e94132a69528)

**Comment by jmarintur** [comment]
Hi @unrue, OpenCV uses BGR image format. You can simply do the following:

```
from cv2 import cvtColor, COLOR_BGR2RGB

for i in range(num_images):
    print("index: ", i, "image: ", inference_dir+'/'+img_list[i])
    im = cv2.imread(inference_dir+'/'+img_list[i])
    # get your image in RGB format
    im = cvtColor(im, COLOR_BGR2RGB)
    dimensions = im.shape
    results = model(im)
    results.save("img_"+str(i)+".jpg", save_dir='out_with_bb') 
    res_list.append(results.pandas().xyxy[0])
    img_dimensions_list.append(dimensions)

```

**Reply by unrue** [reply]
Thanks, but why It happens only on some images?

**Reply by jmarintur** [reply]
Mind sharing these images @unrue ?

**Comment by unrue** [comment]
![003809](https://github.com/ultralytics/yolov5/assets/51996285/a0991c7d-e248-4125-842a-3abaf47ccd42)

The first one I already shared, the second one, is attached (it works with right colors)

**Comment by jmarintur** [comment]
Hi @unrue, I've just gone through the whole process, and I as mentioned before, you need to swap the bands. Here are the outputs I got using the images you shared: 

![image](https://github.com/ultralytics/yolov5/assets/28574023/55d05483-8998-4515-9db8-5e4e0f47d746)

Here is the code: 

```
# image = "305950817-a0991c7d-e248-4125-842a-3abaf47ccd42.jpg"
image = "302577313-32bbc9a5-ca40-4331-abc9-66918effdbeb.jpg"
im = cv2.imread(image)
im = cvtColor(im, COLOR_BGR2RGB) # you could also swap manually the bands, i.e. #im = im[...,[2,1,0]]

results = model(im)
#results.save(image.replace('.jpg','_output.jpt'), save_dir='out_with_bb') 
results.show()
```

**Comment by unrue** [comment]
Ok, thanks!

**Reply by jmarintur** [reply]
@unrue, If it works for you, please remember to select the right answer. Thanks!

**Comment by unrue** [comment]
Actually the cluster where I did inference is offline I'll try as soon as possible.

**Comment by Mactarvish** [comment]
`vis_image_np = src_image_np[..., ::-1]` helps
